Interactive Cli doesn't add references from .csx files.

cli.csx:
```csharp
#r "nuget: Ben.Demystifier, 0.1.1"
#r "nuget: Newtonsoft.Json, *"

using System.Diagnostics;
using Newtonsoft.Json;

static IEnumerable<T> ForEach<T>(this IEnumerable<T> self, Action<T> action)
{
    foreach(var item in self)
    {
        action(item);
    }
    return self;
}

static T Dump<T>(this T self)
{
    try
    {
        Console.WriteLine(JsonConvert.SerializeObject(self, Formatting.Indented));
    }
    catch (System.Exception ex)
    {
        Console.WriteLine(ex.Demystify().ToString());
    }
    return self;
}
```

Console:
```
SlowLogicBoy@Ma-PC MINGW64 ~/Desktop/Tests/Scripts
$ dotnet script
> #load "cli.csx"
C:\Users\SlowLogicBoy\Desktop\Tests\Scripts\cli.csx(5,7): error CS0246: The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or an assembly reference?)
C:\Users\SlowLogicBoy\Desktop\Tests\Scripts\cli.csx(20,27): error CS0103: The name 'JsonConvert' does not exist in the current context
C:\Users\SlowLogicBoy\Desktop\Tests\Scripts\cli.csx(20,61): error CS0103: The name 'Formatting' does not exist in the current context
C:\Users\SlowLogicBoy\Desktop\Tests\Scripts\cli.csx(24,30): error CS1061: 'Exception' does not contain a definition for 'Demystify' and no extension method 'Demystify' accepting a first argument of type 'Exception' could be found (are you missing a using directive or an assembly reference?)
>
```

if I do:
```csharp
#r "nuget: Ben.Demystifier, 0.1.1"
#r "nuget: Newtonsoft.Json, *"
```
in cli and then load that script it works.

--- a/src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs
+++ b/src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs
@@ -12,6 +12,7 @@ using System;
 using System.Collections.Immutable;
 using Dotnet.Script.DependencyModel.NuGet;
 using Dotnet.Script.DependencyModel.Context;
+using System.Diagnostics;
 
 namespace Dotnet.Script.Core
 {
@@ -68,6 +69,7 @@ namespace Dotnet.Script.Core
             {
                 if (_scriptState == null)
                 {
+                    //Debugger.Launch();
                     var sourceText = SourceText.From(input);
                     var context = new ScriptContext(sourceText, CurrentDirectory, Enumerable.Empty<string>(),scriptMode: ScriptMode.REPL, packageSources: _packageSources);
                     await RunFirstScript(context);
@@ -76,6 +78,7 @@ namespace Dotnet.Script.Core
                 {
                     if (input.StartsWith("#r ") || input.StartsWith("#load "))
                     {
+                        Debugger.Launch();
                         var lineRuntimeDependencies = ScriptCompiler.RuntimeDependencyResolver.GetDependencies(CurrentDirectory, ScriptMode.REPL,_packageSources, input).ToArray();
                         var lineDependencies = lineRuntimeDependencies.SelectMany(rtd => rtd.Assemblies).Distinct();
 

--- a/src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs
+++ b/src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs
@@ -1,18 +1,17 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dotnet.Script.DependencyModel.Context;
+using Dotnet.Script.DependencyModel.NuGet;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Scripting.Hosting;
 using Microsoft.CodeAnalysis.Scripting;
 using Microsoft.CodeAnalysis.Scripting.Hosting;
-using System.IO;
 using Microsoft.CodeAnalysis.Text;
-using System.Linq;
-using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis;
-using System.Text;
-using System;
-using System.Collections.Immutable;
-using Dotnet.Script.DependencyModel.NuGet;
-using Dotnet.Script.DependencyModel.Context;
-using System.Diagnostics;
 
 namespace Dotnet.Script.Core
 {
@@ -69,7 +68,6 @@ namespace Dotnet.Script.Core
             {
                 if (_scriptState == null)
                 {
-                    //Debugger.Launch();
                     var sourceText = SourceText.From(input);
                     var context = new ScriptContext(sourceText, CurrentDirectory, Enumerable.Empty<string>(),scriptMode: ScriptMode.REPL, packageSources: _packageSources);
                     await RunFirstScript(context);
@@ -77,8 +75,7 @@ namespace Dotnet.Script.Core
                 else
                 {
                     if (input.StartsWith("#r ") || input.StartsWith("#load "))
-                    {
-                        Debugger.Launch();
+                    {                        
                         var lineRuntimeDependencies = ScriptCompiler.RuntimeDependencyResolver.GetDependencies(CurrentDirectory, ScriptMode.REPL,_packageSources, input).ToArray();
                         var lineDependencies = lineRuntimeDependencies.SelectMany(rtd => rtd.Assemblies).Distinct();
 

--- a/src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs
+++ b/src/Dotnet.Script.Core/Interactive/InteractiveRunner.cs
@@ -75,7 +75,7 @@ namespace Dotnet.Script.Core
                 else
                 {
                     if (input.StartsWith("#r ") || input.StartsWith("#load "))
-                    {                        
+                    {
                         var lineRuntimeDependencies = ScriptCompiler.RuntimeDependencyResolver.GetDependencies(CurrentDirectory, ScriptMode.REPL,_packageSources, input).ToArray();
                         var lineDependencies = lineRuntimeDependencies.SelectMany(rtd => rtd.Assemblies).Distinct();
 

--- a/src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptFilesResolver.cs
+++ b/src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptFilesResolver.cs
@@ -15,11 +15,33 @@ namespace Dotnet.Script.DependencyModel.ProjectSystem
             return result;
         }
 
+        public HashSet<string> GetScriptFilesFromCode(string code)
+        {
+            HashSet<string> result = new HashSet<string>();
+
+            var loadDirectives = GetLoadDirectives(code);
+            foreach (var loadDirective in loadDirectives)
+            {
+                string referencedScript;
+                if (!Path.IsPathRooted(loadDirective))
+                {
+                    referencedScript = Path.GetFullPath((new Uri(Path.Combine(Directory.GetCurrentDirectory(), loadDirective))).LocalPath);
+                }
+                else
+                {
+                    referencedScript = loadDirective;
+                }
+                Process(referencedScript, result);
+            }
+
+            return result;
+        }
+
         private void Process(string csxFile, HashSet<string> result)
         {
             if (result.Add(csxFile))
             {
-                var loadDirectives = GetLoadDirectives(csxFile);
+                var loadDirectives = GetLoadDirectives(File.ReadAllText(csxFile));
                 foreach (var loadDirective in loadDirectives)
                 {
                     string referencedScript;
@@ -37,9 +59,8 @@ namespace Dotnet.Script.DependencyModel.ProjectSystem
             }
         }
 
-        private static string[] GetLoadDirectives(string csxFile)
-        {
-            var content = File.ReadAllText(csxFile);
+        private static string[] GetLoadDirectives(string content)
+        {            
             var matches = Regex.Matches(content, @"^\s*#load\s*""\s*(.+)\s*""", RegexOptions.IgnoreCase | RegexOptions.Multiline);
             List<string> result = new List<string>();
             foreach (var match in matches.Cast<Match>())

--- a/src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptProjectProvider.cs
+++ b/src/Dotnet.Script.DependencyModel/ProjectSystem/ScriptProjectProvider.cs
@@ -1,4 +1,5 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using Dotnet.Script.DependencyModel.Environment;
@@ -27,18 +28,33 @@ namespace Dotnet.Script.DependencyModel.ProjectSystem
 
         public string CreateProjectForRepl(string code, string targetDirectory, string defaultTargetFramework = "net46")
         {
-            var parseresult = _scriptParser.ParseFromCode(code);
+            var scriptFiles = _scriptFilesResolver.GetScriptFilesFromCode(code);
+            targetDirectory = Path.Combine(targetDirectory, "interactive");
+
+            var parseResultFromCode = _scriptParser.ParseFromCode(code);
+            var parseResultFromLoadedFiles = _scriptParser.ParseFromFiles(scriptFiles);
+            var allPackageReferences = new HashSet<PackageReference>();
+
+            foreach (var packageReference in parseResultFromCode.PackageReferences)
+            {
+                allPackageReferences.Add(packageReference);
+            }
+
+            foreach (var packageReference in parseResultFromLoadedFiles.PackageReferences)
+            {
+                allPackageReferences.Add(packageReference);
+            }
 
             targetDirectory = Path.Combine(targetDirectory, "interactive");
             var pathToProjectFile = GetPathToProjectFile(targetDirectory);
             var projectFile = new ProjectFile();
 
-            foreach (var packageReference in parseresult.PackageReferences)
+            foreach (var packageReference in allPackageReferences)
             {
                 projectFile.AddPackageReference(packageReference);
             }
 
-            projectFile.SetTargetFramework(parseresult.TargetFramework ?? defaultTargetFramework);
+            projectFile.SetTargetFramework(parseResultFromCode.TargetFramework ?? defaultTargetFramework);
 
             projectFile.Save(pathToProjectFile);
 
@@ -73,7 +89,7 @@ namespace Dotnet.Script.DependencyModel.ProjectSystem
         {
             _logger.Debug($"Creating project file for {scriptFile}");
             var scriptFiles = _scriptFilesResolver.GetScriptFiles(scriptFile);
-            return CreateProjectFileFromScriptFiles(Path.GetDirectoryName(scriptFile), _scriptEnvironment.TargetFramework, scriptFiles.ToArray());                        
+            return CreateProjectFileFromScriptFiles(Path.GetDirectoryName(scriptFile), _scriptEnvironment.TargetFramework, scriptFiles.ToArray());
         }
 
         private string CreateProjectFileFromScriptFiles(string targetDirectory, string defaultTargetFramework, string[] csxFiles)

--- a/src/Dotnet.Script.Shared.Tests/InteractiveRunnerTestsBase.cs
+++ b/src/Dotnet.Script.Shared.Tests/InteractiveRunnerTestsBase.cs
@@ -230,6 +230,26 @@ namespace Dotnet.Script.Shared.Tests
         }
 
         [Fact]
+        public async Task LoadFileWithNuGetReference()
+        {
+            var pathToFixture = Path.Combine("..", "..", "..", "..", "Dotnet.Script.Tests", "TestFixtures", "InlineNugetPackage", "InlineNugetPackage.csx");
+            var commands = new[]
+            {                
+                $@"#load ""{pathToFixture}""",
+                "using AutoMapper;",
+                "typeof(MapperConfiguration)",
+                "#exit"
+            };
+
+            var ctx = GetRunner(commands);
+            await ctx.Runner.RunLoop();
+
+            var result = ctx.Console.Out.ToString();
+            Assert.Contains("AutoMapper.MapperConfiguration", result);
+        }
+
+
+        [Fact]
         public async Task ResetCommand()
         {
             var commands = new[]

